### DISCUSSION TITLE
Welcome to aws-iot-core-lorawan Discussions!

### DISCUSSION THREAD

**Discussion by asvand** [aws-samples/aws-iot-core-lorawan#8] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by sscaglia** [comment]
Hello:
The deployment seems to fail;

Initiating deployment
=====================
Uploading to samplebinarytransform/39c95cd9236571ae854dc8e600404cb1.template  9316 / 9316  (100.00%)
Error: Failed to create changeset for the stack: samplebinarytransform, An error occurred (ValidationError) when calling the CreateChangeSet operation: Parameter 'EnableNodeJSSupport' must be one of AllowedValues

Is there any setup required?
Thanks

**Reply by asvand** [reply]
Hi sscaglia and thanks for your message! 

Could you please ensure that you use the value "true" or "false" (and not "True" or "False") for the parameter EnableNodeJSSupport as you configure the stack? 

Reason for that ask: there seems to be an issue that sometimes AWS SAM proposes to use the default value "True" instead of "true" as specified in the SAM template. I can not guarantee you that this is the root case in your case, but it is worth a try.

I would be very thankful if you give me a hint if it helped!

**Reply by sscaglia** [reply]
Hi, and thanks for your help.
Actually, I run again from a different machine, where I had aws cli v2 instead of v1 and everything worked... So I suspect that might be the problem... I will re-run in the first machine after upgrading aws cli to v2 and check. I will let you know if I continue with problems. 
Thanks

**Reply by asvand** [reply]
Sure, happy to help, let's get it working! I recommend to either rerun "sam deploy --guided" again and check for right parameter values, as running "sam deploy" will take the existing (possibly wrong) parameter values from file samconfig.toml.

**Reply by sscaglia** [reply]
Hello asvand:

I re-intalled aws (now 2.0) but I continued with the same error...
I checked your observation (True vs true) and YES, this is what was happening. After changing the default (True) by true, I was able to deploy without issues.

Weird is that in the second laptop, I was using the defaults and still seems to be working... I will have to revisit this to investigate.
Anyway, I confirm this is now working.
Thanks

**Reply by asvand** [reply]
Hi sscaglia, first sorry for that inconvenience, this looks to me like a repeating issue in AWS SAM CLI and I will work with our SAM team to resolve that. I am glad that the deployment worked - please be invited to follow-up in case of any questions or feedback on missing features you may have.  Thanks!

**Reply by BrandiDowns** [reply]
@asvand  - Thank you for your suggestion to make sure values are lowercase. Coming back to this page 3 years later, this was my issue and was quickly solved thanks to your reply :)
